OHIF/Viewers
# fix(SR): When loading DICOM SR, only one measurement is shown with no way to show others

### Context

The DICOM SR viewport specified in the description that it was a previous of the measurements, however, the actual
implementation displayed only the first series measurements.  This change creates a temporary display set containing
all images with measurements on them so that the actual display can be shown.

To implement that, fixes were required in the display set service:

* Allow updating display sets when new DICOM instances are added which are applicable to the display set.
* Generate the display sets in O(n log n) time on n images instead of O(n^2) time, making the time to initial images much faster for large display sets.

### Changes & Results

Can now see all the measurements on different series when loading the DICOM SR preview for a multi-series measurement store.

Can save measurements again (was broken)

Time to load on study with 2 900 image series is noticeable faster - but hard to measure.

Added a hanging protocol for SR and for Seg  `@ohif/sr` and `@ohif/seg`

### Testing

* Open up a study with at least 2 series containing images
* Add markup to two different images in different series, and a second image in one of the series
* Save the annotations
* Refresh
* Load the DICOM  SR preview window.  You can see the various measurements on the preview.

* Open up a study with a 2000+ DICOM instance series before
* Open up the same study after, you should see it is faster - I found it noticeable before/after  

Test in the e2e environment with the two URLs:
http://localhost:3000/viewer?StudyInstanceUIDs=1.2.840.113619.2.5.1762583153.215519.978957063.78&hangingProtocolId=@ohif/sr
(this will be blank initially due to a bug in CS3D or perhaps in the dialog service)

http://localhost:3000/viewer?StudyInstanceUIDs=1.3.12.2.1107.5.2.32.35162.30000015050317233592200000046&hangingProtocolId=@ohif/seg

### Checklist

#### PR



- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

#### Public Documentation Updates



- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS: 
- [] "Node version: 
- [] "Browser:
  


[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch
